Command/PopulateCommand.php

```diff
@@ -100,6 +101,81 @@ protected function execute(InputInterface $input, OutputInterface $output)
     }
 
     /**
+     * @param ProviderInterface $provider
+     * @param OutputInterface $output
+     * @param string $input
+     * @param string $type
+     * @param array $options
+     */
+    private function doPopulateType(ProviderInterface $provider, OutputInterface $output, $input, $type, $options)
+    {
+        $loggerClosure = $this->getLoggerClosure($output, $input, $type);
+
+        $provider->populate($loggerClosure, $options);
+    }
+
+    /**
+     * Builds a loggerClosure to be called from inside the Provider to update the command
+     * line.
+     *
+     * @param OutputInterface $output
+     * @param string $index
+     * @param string $type
+     * @return callable
+     */
+    private function getLoggerClosure(OutputInterface $output, $index, $type)
+    {
+        if (!class_exists('Symfony\Component\Console\Helper\ProgressBar')) {
+            $lastStep = null;
+            $current = 0;
+
+            return function ($increment, $totalObjects) use ($output, $index, $type, &$lastStep, &$current) {
+                if ($increment > $totalObjects) {
+                    $increment = $totalObjects;
+                }
+
+                $currentTime = microtime(true);
+                $timeDifference = $currentTime - $lastStep;
+                $objectsPerSecond = $lastStep ? ($increment / $timeDifference) : $increment;
+                $lastStep = $currentTime;
+                $current += $increment;
+                $percent = 100 * $current / $totalObjects;
+
+                $output->writeln(sprintf(
+                    '<info>Populating</info> <comment>%s/%s</comment> %0.1f%% (%d/%d), %d objects/s (RAM: current=%uMo peak=%uMo)',
+                    $index,
+                    $type,
+                    $percent,
+                    $current,
+                    $totalObjects,
+                    $objectsPerSecond,
+                    round(memory_get_usage() / (1024 * 1024)),
+                    round(memory_get_peak_usage() / (1024 * 1024))
+                ));
+            };
+        }
+
+        ProgressBar::setFormatDefinition('normal', " %current%/%max% [%bar%] %percent:3s%%\n%message%");
+        ProgressBar::setFormatDefinition('verbose', " %current%/%max% [%bar%] %percent:3s%% %elapsed:6s%\n%message%");
+        ProgressBar::setFormatDefinition('very_verbose', " %current%/%max% [%bar%] %percent:3s%% %elapsed:6s%/%estimated:-6s%\n%message%");
+        ProgressBar::setFormatDefinition('debug', " %current%/%max% [%bar%] %percent:3s%% %elapsed:6s%/%estimated:-6s% %memory:6s%\n%message%");
```

Comment:
I dont see a better way to provide feedback about what we're doing while the progress bar is running. It'd be nice if the default formats provided a message placeholder, even if it was on the same line.

If the ProgressBar shouldnt use multi line messages because of platform support we should submit a change to the docs.